sds_data_manager/lambda_code/SDSCode/upload_api.py

```diff
@@ -69,55 +103,52 @@ def lambda_handler(event, context):
         return {
             "statusCode": 400,
             "body": json.dumps(
-                "No filename given for upload. "
-                "Please provide a filename "
+                "No filename given for the upload. Please provide a filename "
                 "in the path. Eg. /upload/path/to/file/filename.pkts"
             ),
         }
-    # TODO: Handle other filetypes other than science files
-    #      The current ScienceFilePath only accepts filenames, not the full path
+
     filename = os.path.basename(path_params)
     try:
-        science_file = ScienceFilePath(filename)
-    except ScienceFilePath.InvalidScienceFileError as e:
+        return _spice_file_upload(filename)
+    except imap_data_access.SPICEFilePath.InvalidSPICEFileError:
+        # Not a good SPICE filename, continue on and try science file type next
+        pass
+
+    # Upload for science files, which will catch other filetype errors
+    return _science_file_upload(filename)
+
+
+def _spice_file_upload(filename):
```

Comment:
A quick preference request. Can we move all of these functions above `lambda_handler` function? I got used to having `lambda_handler` function at the bottom and everything that it calls above it.

Reply:
Yes, I moved that down to the bottom of the function. After the previous refactor, the spice and science functions were largely the same, so I actually removed those two helper functions entirely and inlined the code into `lambda_handler`.